nodeScheduler.js

```diff
@@ -0,0 +1,54 @@
+const aws = require('aws-sdk');
+const path = require('path');
+const { db } = require('./server/db');
+
+const s3 = new aws.S3({
+  region: process.env.AWS_REGION,
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+  signatureVersion: 'v4',
+});
+
+const DeleteS3Object = async (imageUrl) => {
+  try {
+    const imageName = path.basename(imageUrl);
+
+    const params = {
+      Bucket: process.env.S3_BUCKET_NAME,
+      Key: imageName,
+    };
+    await s3.deleteObject(params).promise();
+  } catch (error) {
+    // console.log('Error deleting s3 object: ', error);
+  }
+};
+
+const deleteRoutes = async () => {
+  try {
+    await db.query(
+      `DELETE FROM routes WHERE (date < CURRENT_DATE - 15) AND (id IN (SELECT route_id FROM donations WHERE status = 'archived'))`,
```

Comment:
Instead of checking the route id if it's associated with any archived donation, do the converse and make sure that none of the donations associated with route id are not archived already, using the following logic:
`EMPTY(SELECT * FROM donations WHERE route_id = id AND status != 'archived')`
Make sure to test this logic as well before pushing changes.

The motivation for this change is in the very rare case that a route has only some of its donations that have been archived.